rollup.config.js

```diff
@@ -1,37 +1,66 @@
 const commonjs = require('rollup-plugin-commonjs');
-const path = require('path');
 const resolve = require('rollup-plugin-node-resolve');
 const typescript = require('rollup-plugin-typescript');
 const pkg = require('./package.json');
 
-const selfId = path.resolve(__dirname, 'src/iife-self-placeholder.js');
+const globals = {
+  'whatwg-fetch': 'FakeFetch',
+  'fake-xml-http-request': 'FakeXMLHttpRequest',
+  'route-recognizer': 'RouteRecognizer',
+};
+
+const iife = {
+  banner:
+`var Pretender = (function (self) {
+
+  function getModuleDefault(module) {
+    return module.default || module;	
+  }
+
+  var appearsBrowserified = typeof self !== 'undefined' &&
+                            typeof process !== 'undefined' &&
+                            (Object.prototype.toString.call(process) === '[object Object]' ||
+                             Object.prototype.toString.call(process) === '[object process]');
+
+  var RouteRecognizer = appearsBrowserified ? getModuleDefault(require('route-recognizer')) : self.RouteRecognizer;
+  var FakeXMLHttpRequest = appearsBrowserified ? getModuleDefault(require('fake-xml-http-request')) :
+    self.FakeXMLHttpRequest;
+
+  // fetch related ponyfills
+  var FakeFetch = appearsBrowserified ? getModuleDefault(require('whatwg-fetch')) : self.WHATWGFetch;
+`,
+  footer:
+`
+if (typeof module === 'object') {
+  module.exports = Pretender;	
+} else if (typeof define !== 'undefined') {	
+  define('pretender', [], function() {	
+    return Pretender;	
+  });
```

Comment:
This is the AMD module `define` that ember-cli-pretender depends on. Its not spec compliant, but that is intentional. (https://github.com/ember-cli/loader.js/commit/ce27cff63c10380559b49ef9979ca80619bc5fbb)

Reply:
I've refactored this out into an independent JS file for ease of maintenance